core/payments/reservation/reservation_validator_metrics.go

```diff
@@ -0,0 +1,147 @@
+package reservation
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Tracks metrics for the [ReservationPaymentValidator]
+type ReservationValidatorMetrics struct {
+	reservationSymbolCount           prometheus.Histogram
+	reservationInsufficientBandwidth prometheus.Counter
+	reservationQuorumNotPermitted    prometheus.Counter
+	reservationTimeOutOfRange        prometheus.Counter
+	reservationTimeMovedBackward     prometheus.Counter
+	reservationUnexpectedErrors      prometheus.Counter
```

Comment:
should we just bundle these into a single error count? and users can look at logs for error specifics?

Reply:
This gets at the same root issue as [this conversation](https://github.com/Layr-Labs/eigenda/pull/2044#discussion_r2370237624)

Looking at metrics for prod incidents is way better than looking at logs, IMO.

Reply:
I very much disagree. The log has all the info and you'll always end up reading the logs. You'll want to understand WHY a time moved backward for example, which will require context. These metrics by themselves are not useful anyways is my main point.